File: c/5-Arrays/workout/aw-4.c
#include <stdio.h>

int main() {
    int m, n, p, q;
    scanf("%d %d", &m, &n);
    int A[m][n];
    for (int i = 0; i < m; i++)
        {for (int j = 0; j < n; j++){
            scanf("%d", &A[i][j]);}
            printf("\n");
        }
    
    scanf("%d %d", &p, &q);
    int B[p][q];
    for (int i = 0; i < p; i++)
        {    for (int j = 0; j < q; j++){
            scanf("%d", &B[i][j]);}
            printf("\n");
        }
    
    if (n != p) {
        printf("Invalid input\n");
        return 0;
    }
    
    int C[m][q];
     for(int i= 0;i<m;i++)
     {
         for(int j =0;j<q;j++)
         {
             C[i][j] =0;
             for(int k =0;k<n;k++)
             {
                 C[i][j]+=A[i][k]*B[k][j];
             }
         }
     }
     
     
     for(int i =0;i<m;i++)
     {
         for(int j=0;j<q;j++)
         {
             printf("%d ",C[i][j]);
         }
         printf("\n");
     }
     
}

File: java/4-Arrays/workout/aw-4.java
import java.util.Scanner;
public class A{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int a =sc.nextInt();
        int b =sc.nextInt();
        if(a<0||b<0){
            System.out.println("Invalid input");
            return;
        }
            
        int arr[][] =new int[a][b];
        for(int i = 0;i<a;i++)
        {
            for(int j =0;j<b;j++)
            {
                arr[i][j] =sc.nextInt();
            }
        }
        int min =arr[0][0];
        for(int i =0;i<a;i++)
        {
            for(int j =0;j<b;j++)
            {
                if(arr[i][j]<min)
                {
                    min =arr[i][j];
                }
            }
        }
        System.out.println(min);
        
    }
}